src/main/java/com/imagination/cbs/controller/ContractorController.java

```diff
@@ -30,28 +26,19 @@
 	@Autowired
 	private ContractorService contractorService;
 
-	@GetMapping("/{contractorName}")
-	public List<ContractorDto> getContractorsByContractorName(@PathVariable("contractorName") String contractorName){
-		
-		return contractorService.getContractorsByContractorName(contractorName);
-	}
-	
-	@GetMapping("/index")
-	public Page<ContractorIndexDto> getContractorIndexDeatils(
-			@RequestParam(defaultValue = "0") Integer pageNo, 
-            @RequestParam(defaultValue = "10") Integer pageSize,
-            @RequestParam(defaultValue = "contractorName") String sortingField,
-            @RequestParam(defaultValue = "ASC") String sortingOrder){
-		return contractorService.getContractorIndexDeatils(pageNo, pageSize, sortingField, sortingOrder);
-	}
-	
-	@GetMapping("/index/{contractorName}")
-	public Page<ContractorIndexDto> getContractorIndexDeatilsByContractor(@PathVariable("contractorName") String contractorName,
+	@GetMapping("/search")
+	public Page<ContractorEmployeeDto> getContractorEmployeeDetails(
```

Comment:
do we need these many parameters ? can we use some object to club all these ? sonar may complaint for too many params.